drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/commands/actualcommands/AbstractScenarioSimulationCommand.java

```diff
@@ -57,6 +110,24 @@
         }
     }
 
+    protected CommandResult<ScenarioSimulationViolation> setCurrentContext(ScenarioSimulationContext context) {
+        try {
+            final Simulation toRestore = restorableStatus.getSimulation();
+            if (toRestore != null) {
+                final ScenarioSimulationContext.Status originalStatus = context.getStatus().cloneStatus();
+                context.getScenarioSimulationEditorPresenter().getView().setContent(toRestore);
+                context.getScenarioSimulationEditorPresenter().getModel().setSimulation(toRestore);
```

Comment:
I would rather see both of these lines called in the `ScenarioSimulationEditorPresenter`. Just because it is the presenters job to manage the view and model changes.

Reply:
@Rikkola 
I think they are (  context.getScenarioSimulationEditorPresenter() )